lib/liquid/context.rb

```diff
@@ -68,9 +67,22 @@ def handle_error(e, token = nil)
         e.set_line_number_from_token(token)
       end
 
+      output = nil
+
+      if exception_handler
+        result = exception_handler.call(e)
+        case result
+        when Exception
+          e = result
+          e.set_line_number_from_token(token) if e.is_a?(Liquid::Error)
```

Comment:
can you add a test for this line? I think removing it wont break any tests right now, will it?